anchore_engine/clients/grype_wrapper.py

```diff
@@ -231,32 +237,111 @@ def update_grype_db(grype_db_archive_local_file_location: str, version_name: str
             write_lock.release()
 
 
-def get_vulnerabilities(grype_sbom: str) -> json:
+def _get_grype_db_dir():
+    global grype_db_dir
+    return grype_db_dir
+
+
+def _get_grype_db_session():
+    global grype_db_session
+    return grype_db_session
+
+
+def get_current_grype_db_metadata() -> json:
     """
-    Use grype to scan the provided sbom for vulnerabilites.
+    Return the json contents of the metadata file for the in-use version of grype db
     """
-    global grype_db_dir
+    # Get the path to the latest grype_db metadata file
+    latest_grype_db_metadata_file = os.path.join(
+        _get_grype_db_dir(), METADATA_FILE_NAME
+    )
+
+    # Ensure the file exists
+    if not os.path.exists(latest_grype_db_metadata_file):
+        # If not, return None
+        return None
+    else:
+        # Get the contents of the file
+        with open(latest_grype_db_metadata_file) as read_file:
+            try:
+                return json.load(read_file)
+            except JSONDecodeError:
+                logger.error(
+                    "Unable to decode grype_db metadata file into json: %s", read_file
+                )
+                return None
+
+
+def _get_proc_env():
+    # Set grype env variables, including the grype db location
+    grype_env = {
+        "GRYPE_CHECK_FOR_APP_UPDATE": "0",
+        "GRYPE_LOG_STRUCTURED": "1",
+        "GRYPE_DB_AUTO_UPDATE": "0",
+        "GRYPE_DB_CACHE_DIR": "{}".format(_get_grype_db_dir()),
+    }
+    proc_env = os.environ.copy()
+    proc_env.update(grype_env)
+    return proc_env
 
+
+def get_vulnerabilities_for_sbom(grype_sbom: str) -> json:
+    """
+    Use grype to scan the provided sbom for vulnerabilites.
+    """
     # Get the read lock
     read_lock = grype_db_lock.gen_rlock()
     if read_lock.acquire(blocking=False, timeout=60):
         try:
-            # Set grype env variables, including the grype db location
-            grype_env = {
-                "GRYPE_CHECK_FOR_APP_UPDATE": "0",
-                "GRYPE_LOG_STRUCTURED": "1",
-                "GRYPE_DB_AUTO_UPDATE": "0",
-                "GRYPE_DB_CACHE_DIR": "{}".format(grype_db_dir),
-            }
+            # Get env variables to run the grype scan with
+            proc_env = _get_proc_env()
 
-            proc_env = os.environ.copy()
-            proc_env.update(grype_env)
+            # Format and run the command. Grype supports piping in an sbom string, so we need to this in two steps.
+            # 1) Echo the sbom string to std_out
+            # 2) Pipe that into grype
+            pipe_sub_cmd = "echo '{sbom}'".format(
```

Comment:
I had originally had some code in here (https://github.com/anchore/anchore-engine/blob/83cc719ada1469cab438d433f9b6e5796995418b/anchore_engine/clients/grype_wrapper.py#L255) that was going down the road of  allowing grype to be run more broadly, but took it out, primarily because I don't want to add maintenance and testing burden to the team for code we aren't actively using, but also because our use case here is that engine will build and manage grype db. If we also allow it to be run with latest grype db it pulls down itself that has a good chance of introducing confusion.

Reply:
I'll make this a nit then: whenever a method or function is doing many different things, or directly responsible for many things, that creates a problem: it is harder to maintain, harder to reason about, and harder to write tests. 
Writing tests, or writing code that needs tests - and tests in general - is _not a burden_. The more tested code (regardless of amount of usage) the better.